src/Bicep.Decompiler/TemplateDecompiler.cs

```diff
@@ -96,6 +93,22 @@ public static (Uri entrypointUri, ImmutableDictionary<Uri, string> filesToSave)
             return (entryUri, PrintFiles(workspace));
         }
 
+        private static Uri ChangeExtension(Uri prevUri, string newExtension)
+        {
+            var uriString = prevUri.ToString();
+            var finalDot = uriString.LastIndexOf('.');
+            uriString = (finalDot >= 0 ? uriString.Substring(0, finalDot) : uriString) + $".{newExtension}";
+
+            return new Uri(uriString);
+        }
+
+        private static bool HasExtension(Uri uri, string extension)
+        {
+            extension = extension.StartsWith(".") ? extension : $".{extension}";
+
+            return uri.AbsolutePath.EndsWith(extension, StringComparison.OrdinalIgnoreCase);
+        }
```

Comment:
Would be good to put this logic into the `PathHelper` class.

Reply:
Good pt. Move them to `PathHelper`.